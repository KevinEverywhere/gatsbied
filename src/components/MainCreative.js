import React from 'react'
import Assets from './Assets'
import Cameras from './Cameras'
import Entities from './Entities'

class MainCreative extends React.PureComponent {
  constructor(props){
    super();
    this.state={
      loadedContent:null,
      aframeLoaded: null,
    }
  }
  componentDidMount(){
    import('aframe')
      .then((aframe) => {
        if(!!!aframe.components['video-texture']){
          aframe.registerComponent("video-texture", {
            init: function() {
              this.el.addEventListener("click", (e) => {
                console.log(document.querySelector(this.el.getAttribute('src')).paused);
                if(document.querySelector(this.el.getAttribute('src')).paused){
                  document.querySelector(this.el.getAttribute('src')).play()
                }else{
                  document.querySelector(this.el.getAttribute('src')).pause()
                }
              })
            }
          })
        }
        this.setState({
          aframeLoaded: true,
          loadedContent:(
            <main>
              <a-scene cursor="rayOrigin: mouse" embedded style={{width:'100%',height:'100%'}}>
                <Assets />
                <Entities />
                <Cameras />
              </a-scene>
            </main>
          )})
          // aframe.ANIME.timeline({
          //   targets:document.querySelector('#rig'),
          //   position: [650,0,600],
          //   easing: 'easeInOutSine',
          //   endDelay: 1000
          // })
      })
      .catch((error) => console.error(error));
  }
  render() {
    return this.state.loadedContent;
  }
}

export default MainCreative
