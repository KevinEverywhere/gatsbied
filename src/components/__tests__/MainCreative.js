import React from "react"
import renderer from "react-test-renderer"
import MainCreative from "../MainCreative"

describe("MainCreative", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<MainCreative />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
