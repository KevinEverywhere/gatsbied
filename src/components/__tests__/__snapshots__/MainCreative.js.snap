// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainCreative renders correctly 1`] = `null`;
